### ISSUE TITLE
 Issue Generating Reports with ReportCLI

### ISSUE THREAD

**Issue by SR-46** [compomics/peptide-shaker#538]
Hello,

I’m currently working with SearchGui and PeptideShaker on a CPTAC dataset and am encountering some issues. I successfully ran the PeptideShakerCLI, which produced a .psdb file of 66,072 bytes. However, when attempting to generate reports with the ReportCLI, it seems that it doesn't work. (The GUI version also stops at 9% using version 3.0.9, but I don't need it).

When using only -in and -out_reports parameters, the command executes without error, but no report files are generated.
Adding additional parameters like -reports, -report_prefix, -gzip, etc., results in an error.
I’ve updated my Java from version 22.0.1 to 22.0.2, but the issue persists.
I’ve attached my SLURM file content and example dataset for reference. Any assistance or insights into resolving this issue would be greatly appreciated since I have been trying to solve it for days.
Thank you very much.

Path configuration completed.
<CompomicsError>PeptideShaker processing failed. See the PeptideShaker log for details.</CompomicsError>
logs file
Tue Sep 10 12:51:10 CDT 2024: PeptideShaker version 3.0.9.
Memory given to the Java virtual machine: 31675383808.
Total amount of memory in the Java virtual machine: 2164260864.
Free memory: 2144385760.
Java version: 22.0.2.
java.lang.NullPointerException: Cannot invoke "String.trim()" because "input" is null
	at eu.isas.peptideshaker.cmd.ReportCLIInputBean.<init>(ReportCLIInputBean.java:98)
	at eu.isas.peptideshaker.cmd.ReportCLI.main(ReportCLI.java:506)
[slurm.txt](https://github.com/user-attachments/files/16949496/slurm.txt)

**Comment by hbarsnes** [comment]
There was a bug in the ReportCLI gzip option resulting in the exception above. This has now been corrected and a new PeptideShaker version released (v3.0.10). I'm not sure if this fixes everything, so please reopen the issue if this is not the case.

**Comment by SR-46** [comment]
Hi,
Thank you for getting back to me so quickly. The issue I mentioned earlier is now fixed by using the new version. However, I noticed that while I no longer receive errors in the error file, there is error in the output file and the generated reports—aside from the hierarchical one—are coming out as empty files.
Thanks again for your help.
This is the log file

```
Wed Sep 11 15:48:52 UTC 2024: PeptideShaker version 3.0.10.
Memory given to the Java virtual machine: 31675383808.
Total amount of memory in the Java virtual machine: 2164260864.
Free memory: 2138318720.
Java version: 22.0.1-internal.
java.lang.IllegalArgumentException: Modification index out of sequence.
	at com.compomics.util.experiment.identification.features.IdentificationFeaturesGenerator.getModificationSites(IdentificationFeaturesGenerator.java:2320)
	at eu.isas.peptideshaker.export.sections.PsProteinSection.getFeature(PsProteinSection.java:582)
	at eu.isas.peptideshaker.export.sections.PsProteinSection.getFeature(PsProteinSection.java:355)
	at eu.isas.peptideshaker.export.sections.PsProteinSection.writeSection(PsProteinSection.java:261)
	at eu.isas.peptideshaker.export.PSExportFactory.writeExport(PSExportFactory.java:469)
	at eu.isas.peptideshaker.cmd.CLIExportMethods.exportReport(CLIExportMethods.java:670)
	at eu.isas.peptideshaker.cmd.ReportCLI.call(ReportCLI.java:255)
	at eu.isas.peptideshaker.cmd.ReportCLI.main(ReportCLI.java:508)

```
Output file
```
Path configuration completed.
Wed Sep 11 15:48:55 UTC 2024 Exporting. Please Wait...
10% 20% 30%
Wed Sep 11 15:49:00 UTC 2024 An error occurred while exporting the Default Hierarchical Report.
Wed Sep 11 15:49:00 UTC 2024 Exporting. Please Wait...
Wed Sep 11 15:49:00 UTC 2024 Exporting. Please Wait...
Wed Sep 11 15:49:00 UTC 2024 Exporting. Please Wait...
Wed Sep 11 15:49:00 UTC 2024 Exporting. Please Wait...
Wed Sep 11 15:49:00 UTC 2024 Exporting. Please Wait...
Wed Sep 11 15:49:00 UTC 2024 Exporting. Please Wait...
Wed Sep 11 15:49:00 UTC 2024 Exporting. Please Wait...
```

**Comment by hbarsnes** [comment]
There seems to be a problem with mapping one of the PTMs to the peptide sequence. Which PTMs are you using in the search?

Would it also be possible for you to share the data you are trying to export?

**Comment by SR-46** [comment]
Thank you very much. I’m currently using the example from the tutorial with the qExactive01819.mgf file, along with the searchgui_out.zip provided on the website. My plan was to test with the tutorial before moving on to my data.
I’ve attached my SLURM files content for your reference. 
Thank you again.
[slurm1.txt](https://github.com/user-attachments/files/16968819/slurm1.txt)
[slurmReport.txt](https://github.com/user-attachments/files/16968820/slurmReport.txt)

**Comment by hbarsnes** [comment]
There was a bug in the indexing of the PTMs in the protein export. This has now been fixed and a new PeptideShaker version released (v3.0.11). Please note that in order for the fix to work you will have to reload the SearchGUI output in PeptideShaker.

**Comment by SR-46** [comment]
Thank you so much. I could run the example successfully with the new version. I don't know if I should open a new issue or write here. Now that I am using a dataset from CPTAC, I got the attached output file, which shows that the job has been stopped, and the log file doesn't give me much. Could it be due to the large dataset? And if I also want to retrieve all peptides, even if those peptides are not associated with a protein, please tell me which command I need to use. The website says, "To add custom reports, see Export > Identification Features > Reports in PeptideShaker." But I couldn't find it. Thank you very much.
[402slurm.txt](https://github.com/user-attachments/files/16984581/402slurm.txt)
[logs.txt](https://github.com/user-attachments/files/16984582/logs.txt)
[output.txt](https://github.com/user-attachments/files/16984583/output.txt)

**Comment by hbarsnes** [comment]
I sadly cannot learn much about the error from the output either. Is that the only log file you have? Perhaps you can run this test via the GUI and see if that gives you more information?

> "even if those peptides are not associated with a protein". 

Not sure what you mean? All peptides will have to be mapped to a protein, as otherwise there would not be a peptide sequence to compare to the given spectrum. Unless you are talking about de novo sequencing? But in that case PeptideShaker would not be the correct tool to use.

> The website says, "To add custom reports, see Export > Identification Features > Reports in PeptideShaker." But I couldn't find it.

This refers to the main menu in PeptideShaker. You can find it by opening the example dataset and then going to the upper left corner.

**Comment by SR-46** [comment]
Thank you very much. Yes, this is the only log file generated, I tried to change memory but I get the same output. Actually, the GUI version, even for the example in the tutorial, didn't work. It stops in "Importing PSMs from qExactive0819.omx.gz". I tried it with both Mac and Windows. I previously tried SearchGUI and it couldn't handle my data due to memory issue.
I mean the total sequences of peptides before starting be mapped to a protein.
Thank you very much.
